src/Core/src/CommandMapperExtensions.cs

```diff
@@ -26,6 +39,31 @@ void newMethod(TViewHandler handler, TVirtualView view, object? args)
 			}
 		}
 
+		/// <summary>
+		/// Modify a command mapping in place but call the previous mapping if the types do not match.
+		/// </summary>
+		/// <typeparam name="TVirtualView">The cross-platform type.</typeparam>
+		/// <typeparam name="TViewHandler">The handler type.</typeparam>
+		/// <param name="commandMapper">The command mapper in which to change the mapping.</param>
+		/// <param name="key">The name of the command.</param>
+		/// <param name="method">The modified method to call when the command is updated.</param>
+		public static void ModifyMappingWhen<TVirtualView, TViewHandler>(this ICommandMapper<IElement, IElementHandler> commandMapper,
+			string key, Action<TViewHandler, TVirtualView, object?, Action<IElementHandler, IElement, object?>?> method)
+			where TVirtualView : IElement where TViewHandler : IElementHandler
+		{
+			var previousMethod = commandMapper.GetCommand(key);
+
+			void newMethod(IElementHandler handler, IElement view, object? args)
+			{
+				if ((handler is null || handler is TViewHandler) && view is TVirtualView v)
+					method((TViewHandler)handler!, v, args, previousMethod);
+				else
+					previousMethod?.Invoke(handler!, view, args);
+			}
+
+			commandMapper.Add(key, newMethod);
+		}
```

Comment:
This is the new way to do things when you only want to modify/append/prepend if the types match. So instead of replacing the mapper and then doing a check for the type and if it does not match you call base, this does it all for you.

This is the place where you can see it working: https://github.com/dotnet/maui/pull/13836

From:

```cs
EntryHandler.Mapper.ModifyMapping<IEntry, IEntryHandler>("SomeProperty", (handler, view, previous) =>
{
	if (view is Entry entry)
		entry.DoSomething();
	else
		previous.Invoke(handler, view);
});
```

To:

```cs
EntryHandler.Mapper.ModifyMappingWhen<Entry, IEntryHandler>("SomeProperty", (handler, view, previous) =>
{
	view.DoSomething();
});
```

Reply:
Is there a reason to keep the "When" extensions instead of just making the original ones do that check?

if I were to have this currently (before this PR)

```C#
EntryHandler.Mapper.ModifyMapping<Entry, IEntryHandler>("SomeProperty", (handler, view, previous) =>
{
	view.DoSomething();
});
```
Right now it'll throw an exception, right? 

Would a user ever pick one or the other? The behavior of the "When" extension seems like the only valid behavior.

Reply:
OK, I see why I gave it a different name. It is more ambiguous for the developer.

Assume we are modifying the mapping for Editor:

```cs
class EditorHandler {
    IPropertyMapper<IEditor, IEditorHandler> Mapper
}
```

There are 3 ways to modify this:

```cs
// the generics match exactly, so the method is ALWAYS replaced
EditorHandler.Mapper.ModifyMapping<IEditor, IEditorHandler>("Property", MapMethod);

// the generics are more specific (IEditor vs Editor) to this is now the
// new mapper override that will only fire if the type is matching
EditorHandler.Mapper.ModifyMapping<Editor, IEditorHandler>("Property", MapMethod);

// the generics are implicit, so now will fall back to an exact match, or 
// the first one, so we now ALWAYS replace
EditorHandler.Mapper.ModifyMapping("Property", MapMethod);
```

Not sure if this is a pro or a con. Any thoughts?

Reply:
This reads to me like a feature :-) 

```C#
// the generics are more specific (IEditor vs Editor) to this is now the
// new mapper override that will only fire if the type is matching
EditorHandler.Mapper.ModifyMapping<Editor, IEditorHandler>("Property", MapMethod);
```

Is basically what UI frameworks (us) will use. 

Users will most likely continue to use 
```C#
EditorHandler.Mapper.ModifyMapping("Property", MapMethod);
```

or if they get savvy they will use

```C#
EditorHandler.Mapper.ModifyMapping<Editor, EditorHandler>("Property", MapMethod);
```

I feel like, ideally, we'd give users extension methods/paths that were already typed to Controls.

`Entry.Mapper.ModifyMapping("Property"`
or
`EntryHandlerMapper.ModifyMapping<Entry>("Property"`

But we can look at doing that at a later point.